backend/manager/modules/bll/src/main/java/org/ovirt/engine/core/bll/storage/backup/HybridBackupCommand.java

```diff
@@ -135,7 +135,14 @@ protected void executeCommand() {
     @Override
     public boolean performNextOperation(int completedChildCount) {
         restoreCommandState();
-        if (getParameters().getVmBackup().getPhase() == VmBackupPhase.STARTING) {
+        VmBackupPhase phase = getParameters().getVmBackup().getPhase();
+
+        if (getParameters().getVmBackup().isStopped()) {
+            // TODO: if phase == READY, check and image transfer is in progress - wait until it is done
+            return false;
+        }
+
+        if (phase == VmBackupPhase.STARTING) {
```

Comment:
now that I think about it, if we cancel the backup during snapshot creation, we won't create an entry for the checkpoint in the database so we'll have a dangling checkpoint, we should probably either remove it in endSuccessfully or make sure we create the checkpoint entry so we don't lose it, I'd go for the former

Reply:
Okay, will take care of it. Good point

Reply:
After discussing this, the block
```
public boolean performNextOperation(int completedChildCount) {
        ...
        if (phase == VmBackupPhase.STARTING) { ... }
```
should run after snapshot creation, even if the backup was stopped. Moved handling of backup.is_stopped after that, into the phase.READY handling.